### PR TITLE
Issue #77: Replace the class registry with a more robust alternative.

### PR DESCRIPTION
**Pull request by quicksketch**
This PR provides a more robust alternative to the current class registry. Rather than parsing files with regex and then storing the result in a database table, this simply has modules define all their classes via a hook. This alternative prevents the possibility of a corrupted registry, since it is never stored anywhere. It also eliminates database queries when retrieving and updating the registry, though this approach does not have a measurable impact one way or the other on performance.

This also includes a fair amount of code re-arranging, as classes should not be combined with procedural code, otherwise the registry may get triggered upon including these files.

### CODE REVIEW

File: core/includes/bootstrap.classes.inc [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    'FileTransfer' => $path . 'filetransfer/filetransfer.inc',
+    'FileTransferException' => $path . 'filetransfer/filetransfer.inc',
+    'SkipDotsRecursiveDirectoryIterator' => $path . 'filetransfer/filetransfer.inc',
+    'FileTransferChmodInterface' => $path . 'filetransfer/filetransfer.inc',
+    'FileTransferFTP' => $path . 'filetransfer/ftp.inc',
+    'FileTransferFTPExtension' => $path . 'filetransfer/ftp.inc',
+    'FileTransferLocal' => $path . 'filetransfer/local.inc',
+    'FileTransferSSH' => $path . 'filetransfer/ssh.inc',
+  );
+}
```
Comment (jenlampton) [inline core/includes/bootstrap.classes.inc]: no newline at end of file

File: core/includes/install.core.inc [hunk 2]
```diff
@@ -1897,13 +1897,9 @@ function _install_configure_form($form, &$form_state, &$install_state) {
 }
 
 /**
-<<<<<<< HEAD
```
Comment (jenlampton) [inline core/includes/install.core.inc]: looks like there are some merge conflicts in here still

File: core/modules/views/views.tests.info [hunk 3]
```diff
@@ -200,19 +198,19 @@ file = tests/handlers/views_handler_sort.test
...
-description = Tests some basic functions of views.module
+description = Tests some basic functions of views.module.
 group = Views
-file = tests/handlers/views_module.test
+file = tests/views_module.test
 
 [ViewsPagerTest]
 name = Pager
-description = Test the pluggable pager system
+description = Test the pluggable pager system.
```
Comment (jenlampton) [inline core/modules/views/views.tests.info]: I like that you are adding periods at the ends of these test descriptions, but can we also create another separate issue to go through and add periods to the end of all test descriptions, so it's not only correct in the files that you were working on here? :)

**Comment by jenlampton** [comment]
This looks really good.  I commend on a few things that need attention in the PR (merge garbage, etc) but otherwise it's looking good!

File: core/includes/install.core.inc (hunk 2 above)
Comment (quicksketch) [inline core/includes/install.core.inc]: Fixed these in https://github.com/backdrop/backdrop/pull/267.

File: core/includes/bootstrap.classes.inc (hunk 1 above)
Comment (quicksketch) [inline core/includes/bootstrap.classes.inc]: Now added, though Github hasn't collapsed this comment. :)

**Comment by quicksketch** [comment]
I've addressed all the feedback (splitting some problems into separate issues so they've been fixed there). Let's merge this in to get broader user-testing. From a conceptual and automated standpoint we've taken this as far as we can solo.

**Pull request merged**